### ISSUE TITLE
Can't read from text file

### ISSUE THREAD

**Issue by cra0zy** [bridgedotnet/Bridge#3173]
Using the newly implemented IO stuff seems to result in a weird error when trying to read from a file.

### Steps To Reproduce

**test.txt**

```txt
Hello World
```

**App.cs**

```csharp
using System.IO;
using System;

public class Program
{
    public static void Main()
    {
        using (var sr = new StreamReader("test.txt"))
            Console.WriteLine(sr.ReadToEnd());
    }
}
```

### Expected Result

```js
Hello World
```

### Actual Result

```c#
System.Exception: uncaught exception: System.ArgumentException: Value does not fall within the expected range.
ctor@http://localhost/ReadLinesTest/bridge/bridge.js:5063:31
ctor@http://localhost/ReadLinesTest/bridge/bridge.js:5246:13
Read@http://localhost/ReadLinesTest/bridge/bridge.js:30761:27
ReadBuffer@http://localhost/ReadLinesTest/bridge/bridge.js:31773:36
ReadToEnd@http://localhost/ReadLinesTest/bridge/bridge.js:31695:21
Main/button.onclick@http://localhost/ReadLinesTest/bridge/ReadLinesTest.js:21:46
```

**Comment by geoffreymcgill** [comment]
@cra0zy – Thanks for informing us of this defect. We were able to reproduce and a pull request has been submitted with a fix. Once the fix has been reviewed and further testing complete, the patch will be included in the next Bridge release, which should be 16.3.3.

**Comment by fabriciomurta** [comment]
The related pull request has just been merged, the fix will be included in Bridge 16.4.0.
